spec/abci/abci++_basic_concepts.md

```diff
@@ -200,25 +204,27 @@ and an [**Echo**](./abci++_methods.md#echo) method that is used for debugging.
 More details on managing state across connections can be found in the section on
 [Managing Application State](./abci%2B%2B_app_requirements.md#managing-the-application-state-and-related-topics).
 
-## Proposal timeout
+## Proposal related timeouts
 
-Immediate execution requires the Application to fully execute the prepared block
-before returning from `PrepareProposal`, this means that CometBFT cannot make progress
-during the block execution.
-This stands on the consensus algorithm critical path: if the Application takes a long time
-executing the block, the default value of *TimeoutPropose* might not be sufficient
-to accommodate the long block execution time and non-proposer nodes might time
-out and prevote `nil`. The proposal, in this case, will probably be rejected and a new round will be necessary.
+`PrepareProposal` and `ProcessProposal` stand on the consensus algorithm critical path,
```

Comment:
Overall, I like this write-up more than the previous one.

Just on thing: you're introducing `ProcProp` in the reasoning, but the problem is way less serious with `ProcProp` than with `PrepProp`:

* At the beginning of a round, all non-proposers will schedule `TimeoutPropose` immediately
* however, with `ProcProp` the `TimeoutPropose` timeout is only scheduled after having received 2/3 of all prevotes. In practice, since the block tends to take a similar time to run at all validators, when a validator schedules `TimeoutPropose` it is likely to be about to finish its own `ProcProp` execution.

IOW, the _real_ problem is `PrepProp` and `TimeoutPropose`

Reply:
Good point, @sergio-mena. Let me rephrase it to remove ProcProp.

Reply:
> all non-proposers will schedule TimeoutPropose immediately

Actually, that's wrong as we saw on Fri (it's true in the arXiv paper, but not in the code)